### PR TITLE
feat(spans): Align otel attributes with sentry span

### PR DESCRIPTION
**Pull request by jjbayer**
To simplify conversion in the future, align attribute names in Otel's span attributes with keys in Sentry's `span.data`. The old keys in `span.data` remain as legacy alias.

See also https://github.com/getsentry/develop/pull/1230 for the span schema / attribute convention.

ref: https://github.com/getsentry/relay/issues/3278

### CODE REVIEW

File: relay-event-schema/src/protocol/span.rs [hunk 1]
```diff
@@ -290,10 +290,10 @@ pub struct SpanData {
     pub user: Annotated<Value>,
 
     /// Replay ID
-    #[metastructure(field = "replay_id")]
+    #[metastructure(field = "sentry.replay.id", legacy_alias = "replay_id")]
     pub replay_id: Annotated<Value>,
 
-    #[metastructure(field = "sdk.name")]
+    #[metastructure(field = "sentry.sdk.name")]
```
Comment (jjbayer) [inline relay-event-schema/src/protocol/span.rs]: No legacy alias need here, this field was only introduced in https://github.com/getsentry/relay/pull/3456.

File: relay-server/src/services/store.rs [hunk 2]
```diff
@@ -1357,6 +1357,7 @@ struct SpanKafkaMessage<'a> {
     event_id: Option<EventId>,
     #[serde(rename(deserialize = "exclusive_time"))]
     exclusive_time_ms: f64,
+    #[serde(default)]
     is_segment: bool,
```
Comment (jjbayer) [inline relay-server/src/services/store.rs:1363]: When we cannot derive `is_segment`, is should be set to `false`.

File: relay-spans/src/otel_to_sentry_tags.rs [hunk 3]
```diff
@@ -1,34 +0,0 @@
...
-        ("sentry.source", "source"),
-        ("sentry.user.email", "user.email"),
-        ("sentry.user.geo.city", "user.geo.city"),
-        ("sentry.user.geo.country_code", "user.geo.country_code"),
-        ("sentry.user.geo.region", "user.geo.region"),
-        ("sentry.user.ip_address", "user.ip_address"),
-        ("sentry.user.segment", "user.segment"),
-        ("sentry.user.username", "user.username"),
-        ("url.full", "request.url"),
-        ("url.query_string", "request.query_string"),
```
Comment (jjbayer) [inline relay-spans/src/otel_to_sentry_tags.rs:32]: @phacops do you know if any of these mappings was already used for product features? If not, I'd rather remove the mapping and start fresh.

File: relay-event-schema/src/protocol/span.rs [hunk 4]
```diff
@@ -200,11 +200,11 @@ pub struct SpanData {
     pub db_system: Annotated<Value>,
 
     /// The sentry environment.
-    #[metastructure(field = "environment")]
+    #[metastructure(field = "sentry.environment", legacy_alias = "environment")]
```
Comment (jjbayer) [inline relay-event-schema/src/protocol/span.rs:203]: The disadvantage of this rename is that the new key will now be serialized into the payload. But `span.data` is not (yet) forwarded to sentry directly on standalone spans (we only extract some of it into `sentry_tags`, but that happens in relay). Spans as part of transactions are materialized into nodestore, but they don't rely on these particular `span.data` items AFAIK.

File: relay-spans/src/span.rs [hunk 5]
```diff
@@ -169,11 +185,8 @@ pub fn otel_to_sentry_span(otel_span: OtelSpan) -> EventSpan {
             (otel_span.end_time_unix_nano - otel_span.start_time_unix_nano) as f64 / 1e6f64;
     }
 
-    // TODO: This is wrong, a segment could still have a parent in the trace.
-    let is_segment = parent_span_id.is_none().into();
-
```
Comment (jjbayer) [inline relay-spans/src/span.rs:174]: Relying on `set_segment_attributes` instead.

File: relay-spans/src/span.rs [hunk 6]
```diff
@@ -88,79 +94,89 @@ pub fn otel_to_sentry_span(otel_span: OtelSpan) -> EventSpan {
...
-            };
-            if key == "sentry.op" {
-                op = otel_value_to_string(value);
-            } else if key.starts_with("db") {
-                op = op.or(Some("db".to_string()));
-                if key == "db.statement" {
-                    if let Some(statement) = otel_value_to_string(value) {
-                        description = statement;
-                    }
+            match attribute.key.as_str() {
```
Comment (jjbayer) [inline relay-spans/src/span.rs:108]: Sorry for the large diff, I converted this `if` to a `match` expression.

File: relay-spans/src/span.rs [hunk 7]
```diff
@@ -391,4 +408,153 @@ mod tests {
...
+                    "value": {
+                        "stringValue": "my 1st transaction"
+                    }
+                }
+            ]
+        }"#;
+        let otel_span: OtelSpan = serde_json::from_str(json).unwrap();
+        let span_from_otel = otel_to_sentry_span(otel_span);
+
+        // TODO: measurements, metrics_summary
```
Comment (jjbayer) [inline relay-spans/src/span.rs:482]: This will need to be handled in a follow-up.

File: relay-spans/src/span.rs (hunk 6 above)
Comment (Dav1dde) [reply relay-spans/src/span.rs:108]: Large diffs that clean up code make me happy.

File: relay-spans/src/span.rs [hunk 8]
```diff
@@ -169,11 +185,8 @@ pub fn otel_to_sentry_span(otel_span: OtelSpan) -> EventSpan {
             (otel_span.end_time_unix_nano - otel_span.start_time_unix_nano) as f64 / 1e6f64;
     }
 
-    // TODO: This is wrong, a segment could still have a parent in the trace.
-    let is_segment = parent_span_id.is_none().into();
-
     if let (Some(http_method), Some(http_route)) = (http_method, http_route) {
-        description = format!("{} {}", http_method, http_route);
+        description = Some(format!("{} {}", http_method, http_route));
```
Comment (Dav1dde) [inline relay-spans/src/span.rs]:
```suggestion
        description = Some(format!("{http_method} {http_route}"));
```

**Review by Dav1dde — APPROVED** [review]

File: relay-event-schema/src/protocol/span.rs [hunk 9]
```diff
@@ -292,11 +292,11 @@ pub struct SpanData {
     pub user: Annotated<Value>,
 
     /// Replay ID
-    #[metastructure(field = "replay_id")]
+    #[metastructure(field = "sentry.replay.id", legacy_alias = "replay_id")]
     pub replay_id: Annotated<Value>,
 
     /// The sentry SDK (see [`crate::protocol::ClientSdkInfo`]).
-    #[metastructure(field = "sdk.name")]
+    #[metastructure(field = "sentry.sdk.name")]
```
Comment (jjbayer) [inline relay-event-schema/src/protocol/span.rs:299]: No need for an alias here, this field was only introduced in https://github.com/getsentry/relay/pull/3456.

**Pull request merged**